File: solutions/268/01.js
'use strict'; const log = console.log; (async ()=>{

/**
 * @param {number[]} nums
 * @return {number}
 */
const missingNumber = nums => {
  const n = nums.length;
  nums.sort((a, b) => a - b);

  if(nums[n-1] !== n) return n;
  if(nums[0])         return 0;

  for(let i = 1; i < n; i++) {
    const expected = nums[i-1] + 1;
    if(nums[i] !== expected)
      return expected;
  }
};

const log = console.log, f = missingNumber,
  eq = require('assert').strictEqual;

eq(f([]), 0);
eq(f([0]), 1);
eq(f([1]), 0);
eq(f([3,0,1]), 2);
eq(f([9,6,4,2,3,5,7,0,1]), 8);

})();

File: solutions/62/01.py
def f(m, n):
    A = [[1 for j in range(n)] for i in range(m)]

    for i in range(1, m):
        for j in range(1, n):
            A[i][j] = A[i - 1][j] + A[i][j - 1]

    return A[-1][-1]

assert f(2, 3) == 3
assert f(3, 7) == 28
